homeassistant/components/yolink/services.py

```diff
@@ -0,0 +1,68 @@
+"""YoLink services."""
+
+import voluptuous as vol
+from yolink.client_request import ClientRequest
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+
+from .const import (
+    ATTR_REPEAT,
+    ATTR_TARGET_DEVICE,
+    ATTR_TEXT_MESSAGE,
+    ATTR_TONE,
+    ATTR_VOLUME,
+    DOMAIN,
+)
+
+SERVICE_PLAY_ON_SPEAKER_HUB = "play_on_speaker_hub"
+
+
+def async_register_services(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Register services for YoLink integration."""
+
+    async def handle_speaker_hub_play_call(service_call: ServiceCall) -> None:
+        """Handle Speaker Hub audio play call."""
+        if service_call.service == SERVICE_PLAY_ON_SPEAKER_HUB:
```

Comment:
Why is this necessary? The service is linked to this method so the only reason this should be called is if the `SERVICE_PLAY_ON_SPEAKER_HUB` service is requested.

Reply:
Unnecessary code removed